lib/threema/receive/delivery_receipt.rb

```diff
@@ -1,12 +1,46 @@
 # frozen_string_literal: true
 
+require 'threema/util'
+
 class Threema
   module Receive
     class DeliveryReceipt
-      attr_reader :content
+      STATUS_TYPE_BYTE = {
+        received: "\x01".b,
+        read: "\x02".b,
+        explicitly_acknowledged: "\x03".b,
+        explicity_declined: "\x04".b
+      }.freeze
+      UNHEXIFIED_MESSAGE_ID_LENGTH = 8
+
+      attr_reader :content, :status, :timestamp, :message_ids
 
       def initialize(content:, **)
         @content = content
+        return unless type_by(content).present?
+
+        @status = type_by(content)
+        @timestamp = Time.now.utc.to_i
+        @message_ids = extract_message_ids(content)
+      end
+
+      private
+
+      def type_by(content)
+        STATUS_TYPE_BYTE.key(content.slice(0))
+      end
+
+      def extract_message_ids(content)
+        type_removed = content.slice(1, content.length - 1)
+        num_message_ids = content.length / UNHEXIFIED_MESSAGE_ID_LENGTH
+
+        message_ids = []
+        num_message_ids.times do |index|
+          start_index = index.zero? ? index : (index * UNHEXIFIED_MESSAGE_ID_LENGTH)
+          end_index = (index + 1) * UNHEXIFIED_MESSAGE_ID_LENGTH - 1
+          message_ids << Threema::Util.hexify(type_removed[start_index..end_index])
+        end
+        message_ids
```

Comment:
The more I look at it I think this should be done with some sort of `.split` method, because that's what you're doing here, no?

Reply:
https://stackoverflow.com/questions/754407/what-is-the-best-way-to-chop-a-string-into-chunks-of-a-given-length-in-ruby

Reply:
Went with the solution `scan`, this means far fewer lines of code. I'm not sure that it improves understandability though. :shrug:

```rb
message_ids_payload.scan(/.{,#{UNHEXIFIED_MESSAGE_ID_LENGTH}}/).reject(&:empty?).map do |message_id|
  Threema::Util.hexify(message_id)
end
```

Reply:
Also, in the SO post you linked to, they pointed out that this can have a major performance hit for larger strings. Don't think that's really a concern in this case, but just to point it out.